src/transformers/models/ernie/modeling_ernie.py

```diff
@@ -0,0 +1,1819 @@
+# coding=utf-8
+# Copyright 2022 The Google AI Language Team Authors and The HuggingFace Inc. team.
+# Copyright (c) 2018, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""PyTorch ERNIE model."""
+
+
+import math
+import os
+import warnings
+from dataclasses import dataclass
+from typing import List, Optional, Tuple, Union
+
+import torch
+import torch.utils.checkpoint
+from torch import nn
+from torch.nn import BCEWithLogitsLoss, CrossEntropyLoss, MSELoss
+
+from ...activations import ACT2FN
+from ...modeling_outputs import (
+    BaseModelOutputWithPastAndCrossAttentions,
+    BaseModelOutputWithPoolingAndCrossAttentions,
+    CausalLMOutputWithCrossAttentions,
+    MaskedLMOutput,
+    MultipleChoiceModelOutput,
+    NextSentencePredictorOutput,
+    QuestionAnsweringModelOutput,
+    SequenceClassifierOutput,
+    TokenClassifierOutput,
+)
+from ...modeling_utils import PreTrainedModel
+from ...pytorch_utils import (
+    apply_chunking_to_forward,
+    find_pruneable_heads_and_indices,
+    is_torch_greater_than_1_6,
+    prune_linear_layer,
+)
+from ...utils import (
+    ModelOutput,
+    add_code_sample_docstrings,
+    add_start_docstrings,
+    add_start_docstrings_to_model_forward,
+    logging,
+    replace_return_docstrings,
+)
+from .configuration_ernie import ErnieConfig
+
+
+logger = logging.get_logger(__name__)
+
+_CHECKPOINT_FOR_DOC = "nghuyong/ernie-3.0-base-zh"
+_CONFIG_FOR_DOC = "ErnieConfig"
+_TOKENIZER_FOR_DOC = "BertTokenizer"
+
+# # TokenClassification docstring
+# _CHECKPOINT_FOR_TOKEN_CLASSIFICATION = "dbmdz/ernie-large-cased-finetuned-conll03-english"
+# _TOKEN_CLASS_EXPECTED_OUTPUT = (
+#     "['O', 'I-ORG', 'I-ORG', 'I-ORG', 'O', 'O', 'O', 'O', 'O', 'I-LOC', 'O', 'I-LOC', 'I-LOC'] "
+# )
+# _TOKEN_CLASS_EXPECTED_LOSS = 0.01
+#
+# # QuestionAnswering docstring
+# _CHECKPOINT_FOR_QA = "deepset/ernie-base-cased-squad2"
+# _QA_EXPECTED_OUTPUT = "'a nice puppet'"
+# _QA_EXPECTED_LOSS = 7.41
+# _QA_TARGET_START_INDEX = 14
+# _QA_TARGET_END_INDEX = 15
+#
+# # SequenceClassification docstring
+# _CHECKPOINT_FOR_SEQUENCE_CLASSIFICATION = "textattack/nghuyong/ernie-3.0-base-zh-yelp-polarity"
+# _SEQ_CLASS_EXPECTED_OUTPUT = "'LABEL_1'"
+# _SEQ_CLASS_EXPECTED_LOSS = 0.01
```

Comment:
I think we have (at least some) checkpoints for ERNIE 2/3, is this correct? Could we use them here maybe?

Reply:
Yes, now we have the checkpoints for ERNIE1.0 ~3.0, all checkpoint can be seen [here](https://github.com/nghuyong/ERNIE-Pytorch#supported-models)

But these checkpoints are all BertForMaskedLM or BertModel model.
So, I comment _CHECKPOINT_FOR_SEQUENCE_CLASSIFICATION